atlasdb-dbkvs-tests/src/test/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,374 @@
+
+    private KeyValueService keyValueServiceWithNonDefaultPrefix;
+    private NamespaceDeleter namespaceDeleterWithNonDefaultPrefix;
+
+    public OracleNamespaceDeleterIntegrationTest() {
+        super(TRM, TRM);
+    }
+
+    @Before
+    public void before() {
+        dbKeyValueServiceConfig = DbKvsOracleTestSuite.getKvsConfig();
+        oracleDdlConfig = (OracleDdlConfig) dbKeyValueServiceConfig.ddl();
+        executorService = MoreExecutors.newDirectExecutorService();
+        connectionManager = DbKvsOracleTestSuite.getConnectionManager(keyValueService);
+
+        namespaceDeleter = createNamespaceDeleter(
+                oracleDdlConfig,
+                dbKeyValueServiceConfig.connection(),
+                DbKvsOracleTestSuite.getConnectionSupplier(keyValueService),
+                executorService);
+
+        OracleDdlConfig ddlConfigWithNonDefaultPrefix = ImmutableOracleDdlConfig.builder()
+                .from(oracleDdlConfig)
+                .tablePrefix(NON_DEFAULT_TABLE_PREFIX)
+                .overflowTablePrefix(NON_DEFAULT_OVERFLOW_TABLE_PREFIX)
+                .build();
+
+        keyValueServiceWithNonDefaultPrefix =
+                ConnectionManagerAwareDbKvs.create(ImmutableDbKeyValueServiceConfig.builder()
+                        .from(dbKeyValueServiceConfig)
+                        .ddl(ddlConfigWithNonDefaultPrefix)
+                        .build());
+
+        namespaceDeleterWithNonDefaultPrefix = createNamespaceDeleter(
+                ddlConfigWithNonDefaultPrefix,
+                dbKeyValueServiceConfig.connection(),
+                DbKvsOracleTestSuite.getConnectionSupplier(keyValueServiceWithNonDefaultPrefix),
+                executorService);
+    }
+
+    private static NamespaceDeleter createNamespaceDeleter(
+            OracleDdlConfig ddlConfig,
+            ConnectionConfig connectionConfig,
+            ConnectionSupplier connectionSupplier,
+            ExecutorService executorService) {
+        OracleTableNameGetter oracleTableNameGetter = OracleTableNameGetterImpl.createDefault(ddlConfig);
+
+        Function<TableReference, OracleDdlTable> ddlTableFactory = tableReference -> OracleDdlTable.create(
+                tableReference,
+                connectionSupplier,
+                ddlConfig,
+                oracleTableNameGetter,
+                new TableValueStyleCache(),
+                executorService);
+
+        return new OracleNamespaceDeleter(OracleNamespaceDeleterParameters.builder()
+                .tablePrefix(ddlConfig.tablePrefix())
+                .overflowTablePrefix(ddlConfig.overflowTablePrefix())
+                .connectionSupplier(connectionSupplier)
+                .tableNameGetter(oracleTableNameGetter)
+                .oracleDdlTableFactory(ddlTableFactory)
+                .userId(connectionConfig.getDbLogin())
+                .build());
+    }
+
+    @After
+    public void after() {
+        namespaceDeleter.deleteAllDataFromNamespace();
```

Comment:
I do use the namespace deleter itself to clean up, so obviously if it doesn't work, all tests will fail. However, it is the easiest way of cleaning up the state enough for the tests. Another alternative I considered was just forcing the container shuts down and restarts, but that's nontrivial and expensive. Additionally. I could use different users, but then I need to setup the docker container to allow for more users...

Reply:
Which does highlight one flaw of these tests... I don't test across different users.

Reply:
Yes. How hard would it be to add these? I'd recommend just proceeding with this PR and doing them subsequently, given this PR's size.